warehouse/query-core/src/test/java/datawave/query/function/AncestorScanRangeProviderTest.java

```diff
@@ -0,0 +1,51 @@
+package datawave.query.function;
+
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class AncestorScanRangeProviderTest {
+    
+    private final Key docKey = new Key("row", "datatype\0d8zay2.-3pnndm.-anolok");
+    private final Key docKeyField = new Key("row", "datatype\0d8zay2.-3pnndm.-anolok", "FIELD");
+    private final Key docKeyFieldValue = new Key("row", "datatype\0d8zay2.-3pnndm.-anolok", "FIELD\0value");
+    
+    private final Key childDocKey = new Key("row", "datatype\0d8zay2.-3pnndm.-anolok.12.34");
+    private final Key childDocKeyField = new Key("row", "datatype\0d8zay2.-3pnndm.-anolok.12.34", "FIELD");
+    private final Key childDocKeyFieldValue = new Key("row", "datatype\0d8zay2.-3pnndm.-anolok.12.34", "FIELD\0value");
+    
+    private final ScanRangeProvider rangeProvider = new AncestorScanRangeProvider();
```

Comment:
add a grandchild example?

Reply:
Added child and grandchild examples by default